### PR TITLE
ci-search: Increase max-age for results to 28 days

### PR DESCRIPTION
**Pull request by brianmcarey**
**What this PR does / why we need it**:

Some contributors have been looking for more history in ci-search to improve their ability of identifying trends with flaky tests.

Increasing the max age from the default 14 days[1] to 28 days

[1] https://github.com/openshift/ci-search/blob/6356b951587bd1ca473de97e41a2422ae02c7efd/cmd/search/main.go#L72

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/cc @dhiller 

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note

```

### CODE REVIEW

File: github/ci/services/ci-search/patches/statefulset.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ spec:
         - --debug-listen=localhost:6060
         - --config=/etc/ci-search/config.yaml
         - --interval=10m
+        - --max-age=672
```
Comment (dhiller) [inline github/ci/services/ci-search/patches/statefulset.yaml:26]:
```suggestion
        - --max-age=672h
```

**Review by dhiller — COMMENTED** [review]
Agree that this will be very helpful. In case that increase in storage consumption is moderate we could think about going up to 90 days even.

**Comment by brianmcarey** [comment]
> Agree that this will be very helpful. In case that increase in storage consumption is moderate we could think about going up to 90 days even.
> 
> EDIT: in a follow up

Yes sure - I just want to see what impact on performance this increase has first.

**Comment by brianmcarey** [comment]
Rehearsal looks good - https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3800/rehearsal-pull-project-infra-ci-search-deploy-test/1864260661723271168

**Comment by brianmcarey** [comment]
/override pull-project-infra-ci-search-deploy-test 

An update to how the lane was configured was required and the rehearsal for this passed.

**Review by dhiller — APPROVED** [review]
/approve

**Pull request merged**